scripts/segmentation_HALO-20240912a.md

```diff
@@ -0,0 +1,410 @@
+
+# seg16s = hv.VLine(pd.Timestamp("2024-09-12T19:46:30")).opts(color = c1, line_width = lw) #
+# seg16e = hv.VLine(pd.Timestamp("2024-09-12T19:58:55")).opts(color = c1, line_width = lw)
+
+# seg17s = hv.VLine(pd.Timestamp("2024-09-12T20:01:27")).opts(color = c2, line_width = lw) #
+# seg17e = hv.VLine(pd.Timestamp("2024-09-12T20:05:05")).opts(color = c2, line_width = lw)
+```
+
+```python jupyter={"outputs_hidden": true, "source_hidden": true}
+# alt = ds["alt"].hvplot()
+# alt * takeoff * landing * \
+#  seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+# seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e
+```
+
+```python jupyter={"outputs_hidden": true, "source_hidden": true}
+# heading = ds["heading"].hvplot()
+# heading * takeoff * landing * \
+#  seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+#  seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e
+```
+
+```python jupyter={"outputs_hidden": true, "source_hidden": true}
+# roll = ds["roll"].hvplot()
+# roll * takeoff * landing * \
+#  seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+#  seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e
+```
+
+## Segments
+
+defined as a tuple of time slice (`start`, `end`) , segment `kind`, `name`, `remarks`.
+
+* in case of irregularities within a circle, 1 sec before the first and after the last sonde are chosen as start and end times
+* use the list of `remarks` to state any deviations, also with respective times
+
+Alternatively, you can also define the segments as dictionaries which also allows to add further attributes to single segments, e.g. a `radius` to a `circle` segment. At the end of the following code block all segments will be normalized by the `parse_segments` function.
+
+```python
+seg1 = (
+    slice("2024-09-12T11:33:02", "2024-09-12T11:46:57"),
+    ["straight_leg", "ascent"],
+    "eastward ascending ferry",
+)
+
+seg2 = (
+    slice("2024-09-12T11:52:44", "2024-09-12T12:01:28"),
+    ["straight_leg", "ascent"],
+    "eastward ascending ferry",
+
+)
+
+seg3 = (
+    slice("2024-09-12T12:01:28", "2024-09-12T12:35:09"),
+    ["straight_leg"],
+    "long eastward ferry",
+)
+
+seg4 = (
+    slice("2024--09-12T12:36:26", "2024-09-12T13:02:59"),
+    ["straight_leg"],
+    "leg to first circle"
+)
+
+seg5 = (
+    slice("2024-09-12T13:06:18", "2024-09-12T14:02:18"),
+    ["clockwise eastern circle"],
+    ["sonde failures"]
+)
```

Comment:
I'd suggest a few things here: slightly later start of the circle as the current one would start at a roll angle of 2 deg deviation from the circle roll angle. The attribute (kind) should be `circle`. I'd leave out the remark on sonde failures as we did not finish the dropsonde QC yet and the numbers of usable sondes might change.
```suggestion
seg5 = (
    slice("2024-09-12T13:06:53", "2024-09-12T14:02:18"),
    ["circle"],
    "clockwise eastern circle",
)
```